metadata-ingestion/src/datahub/ingestion/source/sql/presto_on_hive.py

```diff
@@ -62,7 +76,32 @@ class PrestoOnHiveConfig(BasicSQLAlchemyConfig):
         description="Host URL and port to connect to. Example: localhost:3306",
     )
     scheme: str = Field(default="mysql+pymysql", description="", exclude=True)
+    metastore_db_name: Optional[str] = Field(
+        default=None,
+        description="""Name of the Hive metastore's database (usually: metastore).
+        For backwardcompatibility, if this field is not provided, the 'database' field will be used.
+        If both the 'database' and 'metastore_db_name' fields are set then the 'database'
+        field will be used to filter the hive/presto/trino database"""
+        )
+    mode: PrestoOnHiveConfigMode = Field(
+        default=PrestoOnHiveConfigMode.presto_on_hive,
+        description="""The ingested data will be stored under this platform.
+        Valid options: [presto-on-hive, trino, presto, hive]""",
+    )
+
 
+    def get_sql_alchemy_url(self, uri_opts: Optional[Dict[str, Any]] = None) -> str:
+        if not ((self.host_port and self.scheme) or self.sqlalchemy_uri):
+            raise ValueError("host_port and schema or connect_uri required.")
+
+        return self.sqlalchemy_uri or make_sqlalchemy_uri(
+            self.scheme,  # type: ignore
```

Comment:
why is `# type: ignore` needed here?

Reply:
it doesn't, copied over the function from `sql_common` and it was there.